### ISSUE TITLE
Package container-interop/container-interop is abandoned

### ISSUE THREAD

**Issue by enumag** [laminas/laminas-validator#70]
When installing laminas-validator composer reports this:

```
Package container-interop/container-interop is abandoned, you should avoid using it. Use psr/container instead.
```

**Comment by weierophinney** [comment]
This can be safely ignored.

We depend on laminas-servicemanager in this package, as we provide a plugin manager implementation based off that package. That package currently supports both container-interop and PSR-11. We are working to remove the container-interop dependency, but it _does_ require a BC break, which also affects anybody implementing its current interfaces, as they still typehint on container-interop. As such, please be patient while we figure out the best way to do this.

**Comment by Aerendir** [comment]
This cannot be safely ignored anymore.

Due to this package, my CI breaks.

https://github.com/Aerendir/bundle-features/runs/4362660359?check_suite_focus=true

In the bundle linked, I use a component I wrote that uses `laminas/uri` that, in turn, uses this package `laminas/validator`.

The use of `container-interop/container-interop` makes impossible to have green tests on PHP 8.0.

If you need more info about this, please, let me know.

**Comment by Ocramius** [comment]
> This cannot be safely ignored anymore.
 > 
 > Due to this package, my CI breaks.
 > 
 > https://github.com/Aerendir/bundle-features/runs/4362660359?check_suite_focus=true

The error seems unrelated to `laminas/laminas-validator`, and instead is related to the major version bump to `symfony/*:^6`, which is another/separate problem.

In fact, `container-interop/container-interop` seems totally unrelated there.

**Comment by Aerendir** [comment]
It is not related to `laminas/validator`, but to the `container-interop/container-interop` package that is required by `laminas/validator`.

Maybe the problem arose with SF6, but the root cause is the use of an abandoned package, isn't it?

**Comment by Ocramius** [comment]
As of the error message:

```
PHP Fatal error:  Declaration of Symfony\Component\DependencyInjection\ContainerInterface::has($id) must be compatible with Psr\Container\ContainerInterface::has(string $id): bool in /home/runner/work/bundle-features/bundle-features/vendor/symfony/dependency-injection/ContainerInterface.php on line 63
Fatal error: Declaration of Symfony\Component\DependencyInjection\ContainerInterface::has($id) must be compatible with Psr\Container\ContainerInterface::has(string $id): bool in /home/runner/work/bundle-features/bundle-features/vendor/symfony/dependency-injection/ContainerInterface.php on line 63
```

... no, the root cause is in `symfony/dependency-injection` not declaring that it only supports [`psr/container:^2`](https://github.com/php-fig/container/releases/tag/2.0.0):

https://github.com/symfony/dependency-injection/blob/4dae086ee9bda1e2d0fdbc19a4a30ea49b0e3c7c/composer.json#L20

I suggest you report an issue there.

**Comment by Aerendir** [comment]
Ok, I understand your point of view and I'm going to open an issue there.

But this doesn't solves the problem, as if the package is installed somewhere that requires the use of `symfony/dependency-injection:^6`, then the installation will fail due to the declared constraint, doesn't it?

So, basically, it will be impossible to move to SF6 and at the same time continue to use `laminas/validator`.

**Comment by Ocramius** [comment]
> So, basically, it will be impossible to move to SF6 and at the same time continue to use `laminas/validator`.

Should be possible, based on outcome of https://github.com/container-interop/container-interop/pull/97

Not sure about `container-interop` supporting `psr/container:^2`, but that's a different story too. In practice, BC breaks have to be introduced **somewhere**, and that is worse than allowing an upgrade path to latest releases in general: stable systems over shiny new things.

**Comment by gsteel** [comment]
Closed via #129
